torch/csrc/jit/frontend/convert_to_ssa.cpp

```diff
@@ -93,7 +93,7 @@ struct ControlFlowLoadStores {
     for (const auto& x : mutated_variables) {
       auto true_type = true_vars->findInAnyFrame(x);
       auto false_type = false_vars->findInAnyFrame(x);
-      auto unified = unifyTypes(true_type, false_type);
+      auto unified = unifyTypes(true_type, false_type, true);
```

Comment:
Nit:
```suggestion
      auto unified = unifyTypes(true_type, false_type, /*default_to_union*=/ true);
```

If we allow unifying to Union, then I think `unifyTypes` would never fail, which means line 97-99 wouldn't be necessary